packages/connect/src/types/device.ts

```diff
@@ -34,6 +34,20 @@ export enum FirmwareType {
 // key = coin shortcut lowercase (ex: btc, eth, xrp) OR field declared in coins.json "supportedFirmware.capability"
 export type UnavailableCapabilities = { [key: string]: UnavailableCapability };
 
+export type FirmwareRevisionCheckResult =
+    | {
+          success: true;
+          error: null;
+      }
+    | {
+          success: false;
+          error: 'revision-mismatch' | 'firmware-version-unknown' | 'firmware-version-unknown';
+      }
+    | {
+          success: null;
```

Comment:
i would not step out from the convention `success: boolean` that we use everywhere else.

im not a big fan of using `null` as a meaningful value and distinguishing between `null` and `undefined` because after a while nobody remembers why it's done like that

i think this could be be done like:

```
{
   success: false,
   error: 'cannot-perform-check-offline'
}
```

and evaluated in SuiteBanners and suite Preloader using specific error literal

```
} else if (
    !isFirmwareRevisionCheckDisabled &&
    device?.authenticityChecks.firmwareRevision?.error === 'cannot-perform-check-offline' 
) {

```

Reply:
Good point :+1:

Reply:
https://github.com/trezor/trezor-suite/pull/13495/commits/7b2d5ae513b9c6c2be44bfeb57e4646ef54f8ae2